client/ayon_shotgrid/plugins/publish/integrate_shotgrid_version.py

```diff
@@ -85,7 +105,7 @@ def process(self, instance):
             })
 
         # If there's no data to set/update, skip creation of SG version
-        if not found_reviewable:
+        if not found_reviewable or "workfile" in f"{instance.data['productName']}":
```

Comment:
why not just checking `if not data_to_update`? check our fork https://github.com/fabiaserra/ayon-shotgrid/blob/release/alkemyx/client/ayon_shotgrid/plugins/publish/integrate_shotgrid_version.py#L112